streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -111,6 +111,8 @@ public void initialize() {
             try {
                 partitions = streamsProducer.partitionsFor(topic);
             } catch (final KafkaException e) {
+                // TODO: KIP-572 need to handle `TimeoutException`
+                // -> should we throw a `TaskCorruptedException` for this case to reset the task and retry (including triggering `task.timeout.ms`) ?
```

Comment:
This is an open question. Input is welcome. I would like to tackle it in a follow up PR.